Dockerfile

```diff
@@ -1,31 +1,53 @@
-FROM node:12-alpine
-WORKDIR /app
-RUN apk add --no-cache git
-RUN chown node:node /app
-COPY package.json .
-USER node
-RUN npm install
-COPY webpack* tsconfig.json ./
-COPY src src
-RUN npm run build
-
-FROM golang:1.14-buster
-WORKDIR /app
-RUN chown nobody /app
-RUN go get -u github.com/gobuffalo/packr/packr
-COPY go.mod go.sum ./
-RUN go mod download
-COPY --from=0 /app/build build
-COPY .git .git
-COPY res res
-COPY server server
-COPY main.go .
-RUN packr build -ldflags "-X main.gitDescribe=$(git describe --always --tags)" -o peer-calls main.go
-
-FROM debian:buster-slim
-WORKDIR /app
-COPY --from=1 /app/peer-calls .
-USER nobody
-EXPOSE 3000
+FROM node:12-alpine as frontend
+
+# Add dependency instructions and fetch node_modules
+COPY package.json package-lock.json /src/
+wORKDIR /src
+
+RUN set -ex \
+ && apk add --no-cache \
+      git \
+ && npm ci
+
+# Add the application itself
+COPY ./ /src/
+
+RUN set -ex \
+ && npm run build
+
+
+FROM golang:alpine as server
+
+ENV CGO_ENABLED=0
+
+RUN set -ex \
+ && apk add --no-cache \
+      git \
+ && GOPATH=/usr/local go get -u github.com/gobuffalo/packr/packr
+
+# Add dependencies into mod cache
+COPY go.mod go.sum /src/
+wORKDIR /src
```

Comment:
This is a total nitpick, but noticed the `w` is lowercase, here and in the node build above.

Reply:
Not sure how that happened but thanks for mentioning! (Would have needed to do another PR to fix that as that bothers me a lot too… :see_no_evil:)